addons/godot-xr-tools/functions/movement_footstep.gd

```diff
@@ -0,0 +1,186 @@
+tool
+class_name XRToolsMovementFootstep
+extends XRToolsMovementProvider
+
+## Movement provider order
+export var order : int = 1001
+
+# on ground
+var ground_only : bool = false
+# audio holder path
+var audio : XRToolsSurfaceAudio
+# last surface
+var on_default : bool = false
+var on_fabric : bool = false
+var on_grass : bool = false
+var on_metal : bool = false
+var on_mud : bool = false
+var on_puddle : bool = false
+var on_sand : bool = false
+var on_snow : bool = false
+var on_stone : bool = false
+var on_tile : bool = false
+var on_water : bool = false
+var on_wood : bool = false
+# step time and rate
+var step_rate = 0.5
+var step_time = 0.0
+
+# Last on_ground state of the player
+var _old_on_ground := true
+
+# Previous velocity
+var _previous_velocity : Vector3 = Vector3.ZERO
+var control_velocity := Vector3.ZERO
+
+# Some value indicating the player wants to walk at a moderate speed
+const WALK_SOUND_THRESHOLD := 0.3
+# Add support for is_class on XRTools classes
+func is_class(name : String) -> bool:
+	return name == "XRToolsMovementFootstep" or .is_class(name)
+
+func _ready():
+	# Set as always active
+	is_active = true
+	var player_body := XRToolsPlayerBody.find_instance(self)
+	player_body.connect("player_jumped", self, "_on_player_jumped")
+
+func physics_movement(_delta: float, player_body: XRToolsPlayerBody, disabled: bool):
+	var footstep_holder := XRToolsFootstepHolder.find_instance(self)
```

Comment:
find_instance is an incredibly expensive operation. This will search for it repetitively on every physics frame, and instead the node should probably be cached in the _ready function and then used in the physics_movement function.

Reply:
indeed, thx for pointing that out.
it should now get var footstep_holder onready and the double call of var footstep_holder was removed as well.
previously i was calling it again inside the _on_player_jumped() function and that drained ressources too.

regarding the code comments, i added as suggested and the snake case naming, changed as suggested